packages/x-codemod/src/v7.0.0/pickers/rename-slots-types/index.ts

```diff
@@ -0,0 +1,138 @@
+  DateTimePickerSlotsComponentsProps: 'DateTimePickerSlotProps',
+  DesktopDatePickerSlotsComponent: 'DesktopDatePickerSlots',
+  DesktopDatePickerSlotsComponentsProps: 'DesktopDatePickerSlotProps',
+  DesktopDateTimePickerSlotsComponent: 'DesktopDateTimePickerSlots',
+  DesktopDateTimePickerSlotsComponentsProps: 'DesktopDateTimePickerSlotProps',
+  DesktopTimePickerSlotsComponent: 'DesktopTimePickerSlots',
+  DesktopTimePickerSlotsComponentsProps: 'DesktopTimePickerSlotProps',
+  DigitalClockSlotsComponent: 'DigitalClockSlots',
+  DigitalClockSlotsComponentsProps: 'DigitalClockSlotProps',
+  ExportedPickersLayoutSlotsComponent: 'ExportedPickersLayoutSlots',
+  ExportedPickersLayoutSlotsComponentsProps: 'ExportedPickersLayoutSlotProps',
+  MobileDatePickerSlotsComponent: 'MobileDatePickerSlots',
+  MobileDatePickerSlotsComponentsProps: 'MobileDatePickerSlotProps',
+  MobileDateTimePickerSlotsComponent: 'MobileDateTimePickerSlots',
+  MobileDateTimePickerSlotsComponentsProps: 'MobileDateTimePickerSlotProps',
+  MobileTimePickerSlotsComponent: 'MobileTimePickerSlots',
+  MobileTimePickerSlotsComponentsProps: 'MobileTimePickerSlotProps',
+  MultiSectionDigitalClockSlotsComponent: 'MultiSectionDigitalClockSlots',
+  MultiSectionDigitalClockSlotsComponentsProps: 'MultiSectionDigitalClockSlotProps',
+  PickersCalendarHeaderSlotsComponent: 'PickersCalendarHeaderSlots',
+  PickersCalendarHeaderSlotsComponentsProps: 'PickersCalendarHeaderSlotProps',
+  PickersLayoutSlotsComponent: 'PickersLayoutSlots',
+  PickersLayoutSlotsComponentsProps: 'PickersLayoutSlotProps',
+  StaticDatePickerSlotsComponent: 'StaticDatePickerSlots',
+  StaticDatePickerSlotsComponentsProps: 'StaticDatePickerSlotProps',
+  StaticDateTimePickerSlotsComponent: 'StaticDateTimePickerSlots',
+  StaticDateTimePickerSlotsComponentsProps: 'StaticDateTimePickerSlotProps',
+  StaticTimePickerSlotsComponent: 'StaticTimePickerSlots',
+  StaticTimePickerSlotsComponentsProps: 'StaticTimePickerSlotProps',
+  TimeClockSlotsComponent: 'TimeClockSlots',
+  TimeClockSlotsComponentsProps: 'TimeClockSlotProps',
+  TimePickerSlotsComponents: 'TimePickerSlots',
+  TimePickerSlotsComponentsProps: 'TimePickerSlotProps',
+  DateRangeCalendarSlotsComponent: 'DateRangeCalendarSlots',
+  DateRangeCalendarSlotsComponentsProps: 'DateRangeCalendarSlotProps',
+  DateRangePickerSlotsComponents: 'DateRangePickerSlots',
+  DateRangePickerSlotsComponentsProps: 'DateRangePickerSlotProps',
+  DesktopDateRangePickerSlotsComponent: 'DesktopDateRangePickerSlots',
+  DesktopDateRangePickerSlotsComponentsProps: 'DesktopDateRangePickerSlotProps',
+  MobileDateRangePickerSlotsComponent: 'MobileDateRangePickerSlots',
+  MobileDateRangePickerSlotsComponentsProps: 'MobileDateRangePickerSlotProps',
+  StaticDateRangePickerSlotsComponent: 'StaticDateRangePickerSlots',
+  StaticDateRangePickerSlotsComponentsProps: 'StaticDateRangePickerSlotProps',
+
+  // Next elements are internal types
```

Comment:
Nitpick: These would probably end up being removed when finalizing the PR before merging? 🤔 
Would you also consider adding this transformation to the `v7/preset-safe`? 🤔

Reply:
We should probably work to remove those imports which are a bad habit
But since they are probably used, I'm fine handling them in the codemod :+1: